### PR TITLE
Image download/provision support

### PR DESCRIPTION
**Pull request by pathuang-sia**
Passing down urn and cloudinitdatasource from moc-operator to cloud agent

### CODE REVIEW

File: services/compute/compute.go [hunk 1]
```diff
@@ -666,6 +666,8 @@ type GalleryImageIdentifier struct {
 	Offer *string `json:"offer,omitempty"`
 	// Sku - The name of the gallery Image Definition SKU.
 	Sku *string `json:"sku,omitempty"`
+	// Version - Specifies the version of the platform image or marketplace image used to create the virtual machine.
+	Version *string `json:"version,omitempty"`
```
Comment (madhanrm) [inline services/compute/compute.go:670]: This property doesnt exist in Azure sdk. Why are we introducing it here
https://github.com/Azure/azure-sdk-for-go/blob/main/services/compute/mgmt/2021-11-01/compute/models.go#L7900

File: services/compute/compute.go [hunk 2]
```diff
@@ -753,8 +755,6 @@ type GalleryImageProperties struct {
 	OsType OperatingSystemTypes `json:"osType,omitempty"`
 	// OsState - This property allows the user to specify whether the virtual machines created under this image are 'Generalized' or 'Specialized'. Possible values include: 'Generalized', 'Specialized'
 	OsState OperatingSystemStateTypes `json:"osState,omitempty"`
-	// HyperVGeneration - The hypervisor generation of the Virtual Machine. Applicable to OS disks only. Possible values include: 'V1', 'V2'
```
Comment (madhanrm) [inline services/compute/compute.go:756]: Any reason to remove this comment?

File: services/compute/compute.go [hunk 3]
```diff
@@ -769,6 +769,10 @@ type GalleryImageProperties struct {
 	ContainerName *string `json:"containername,omitempty"`
 	//Type of source of gal image (sfs/http/local)
 	SourceType common.ImageSource `json:"sourceType,omitempty"`
+	// CloudInitDataSource - READ-ONLY; The cloud init data source to be used with the image. [NoCloud, Azure]. Default Value – NoCloud. For marketplace images it will be Azure.
+	CloudInitDataSource common.CloudInitDataSource `json:"cloudInitDataSource,omitempty"`
+	// DownlaodStatus - Current download status of the image
+	DownloadStatus common.DownloadStatus `json:"downloadStatus,omitempty"`
```
Comment (madhanrm) [inline services/compute/compute.go:775]: Try to add DownloadStatus into Status key value map
It doesnt make sense here

File: services/compute/compute.go (hunk 1 above)
Comment (madhanrm) [reply services/compute/compute.go:670]: The Version is specific to http download from market place. 
Check how we do this using SFSImageProperties. Either use httpImageProperties or define a new struct and use it the same way as SfsImageProperties and pass it along as a string.

Try not to introduce params that would be specific to a type

Comment (madhanrm) [reply services/compute/compute.go:670]: And Please document that changes in the design md

File: services/compute/compute.go (hunk 2 above)
Comment (pathuang-sia) [reply services/compute/compute.go:756]: The value will be added in this PR: https://github.com/microsoft/moc-sdk-for-go/pull/96/files. I'll remove this change.

Comment (madhanrm) [reply services/compute/compute.go:756]: still see this change. Can you remove this change?

File: services/compute/compute.go (hunk 3 above)
Comment (madhanrm) [reply services/compute/compute.go:781]: Should this be DownloadStatus or upload status - since we are uploading the image to MOC

Comment (manoj-kadam) [reply services/compute/compute.go]: It's tricky. In doing that it's also downloading the image from Azure Marketplace. In the ARM API it's proposed as DownloadStatus.

Comment (pathuang-sia) [reply services/compute/compute.go]: I'll move downloadstatus under status but will need to modify moc/pkg/status to pass it down. I'll link the pr here once ready.

Comment (madhanrm) [reply services/compute/compute.go]: @pathuang-sia  - I think the ask was different and Manoj answerd it. Thanks.

Comment (pathuang-sia) [reply services/compute/compute.go]: https://github.com/microsoft/moc/pull/116

**Review by madhanrm — APPROVED** [review]

**Pull request merged**